File: services/ai-runtime/app/api/health.py
from datetime import datetime, timezone

STARTED_AT = datetime.now(timezone.utc)

async def healthz() ->dict:
    uptime_seconds = round(
        (datetime.now(timezone.utc) - STARTED_AT).total_seconds(),
        3,
    )

    return {
        "status":"ok",
        "service":"agentflow-ai-runtime",
        "uptime_seconds":uptime_seconds
    }

async def readyz() ->dict:
    return {
        "status":"ready",
        "service":"agentflow-ai-runtime",
        "checks":{
            "runtime":"ok"
        }
    }

File: services/api/internal/handler/health.go
package handler

import (
	"net/http"
	"time"

	"github.com/gin-gonic/gin"
)

type HealthHandler struct {
	startedAt time.Time
}

func NewHealthHandler() *HealthHandler {
	return &HealthHandler{startedAt: time.Now()}
}

func (h *HealthHandler) Healthz(c *gin.Context) {
	c.JSON(http.StatusOK, gin.H{
		"status":  "ok",
		"service": "agentflow-api",
		"uptime":  time.Since(h.startedAt).String(),
	})
}

func (h *HealthHandler) Readyz(c *gin.Context) {
	c.JSON(http.StatusOK, gin.H{
		"status":  "ready",
		"service": "agentflow-api",
		"checks": gin.H{
			"api": "ok",
		},
	})
}
